src/framecontainer/viewerParameters.ts

```diff
@@ -0,0 +1,74 @@
+export interface ViewerParameters {
+  cap?: ViewerCapability[];
+  origin?: string;
+  cid?: boolean;
+  csi?: boolean;
+  dialog?: boolean;
+  development?: boolean;
+  history?: boolean;
+  horizontalScrolling?: boolean;
+  log?: boolean;
+  paddingTop?: number;
+  prerenderSize?: number;
+  p2r?: boolean;
+  referrer?: string;
+  storage?: boolean;
+  viewerUrl?: string;
+  webview?: boolean;
+  highlight?: string;
+  messagingToken?: string;
+  viewportType?: ViewerViewportType;
+  visibilityState?: ViewerVisibilityState;
+}
+
+export enum ViewerCapability {
+  A2A = 'a2a',
+  CLIENT_ID_SERVICE = 'cid',
+  ERROR_REPORTER = 'errorReporter',
+  FRAGMENT = 'fragment',
+  HANDSHAKEPOLL = 'handshakepoll',
+  NAVIGATE_TO = 'navigateTo',
+  REPLACE_URL = 'replaceUrl',
+  SWIPE = 'swipe',
+  VIEWER_RENDER_TEMPLATE = 'viewerRenderTemplate',
+  XHR_INTERCEPTOR = 'xhrInterceptor',
+}
+
+export enum ViewerViewportType {
+  NATURAL = 'natural',
+  NATURAL_IOS_EMBED = 'natural-ios-embed',
+}
+
+export enum ViewerVisibilityState {
+  INACTIVE = 'inactive',
```

Comment:
these values correspond to the values here: https://github.com/ampproject/amphtml/blob/54087c3320bda3d32be56ef4d3c312007b85dec6/src/visibility-state.js

If the values change in the AMP project or new values are added they need to propagate here. We need to generate a ts file from those values for usage in this project. Same for any properties defined here, that originated and is used by runtime in AMP. We can work together on this.

Reply:
Good point. If there's a way to not maintain these types in two places, that'd be awesome. I thought about publishing something like `@types/amp-viewer` and moving this file there, but that doesn't solve the problem of keeping it in sync with the AMP repo.